pyhms/sprout/sprout_filters.py

```diff
@@ -109,22 +105,21 @@ def __init__(self, limit: int) -> None:
 
     def __call__(
         self,
-        candidates: Dict[AbstractDeme, Tuple[Dict[str, float], List[Individual]]],
+        candidates: dict[AbstractDeme, DemeCandidates],
         tree,
-    ) -> Dict[AbstractDeme, Tuple[Dict[str, float], List[Individual]]]:
+    ) -> dict[AbstractDeme, DemeCandidates]:
         for level in range(len(tree.levels[:-1])):
             level_demes = [deme for deme in candidates.keys() if deme.level == level]
-            level_candidates = [candidate for deme in level_demes for candidate in candidates[deme][1]]
+            level_candidates = [candidate for deme in level_demes for candidate in candidates[deme].individuals]
             level_candidates.sort(key=lambda ind: ind.fitness)
             currently_active_level_below = len([deme for deme in tree.levels[level + 1] if deme.is_active])
             if currently_active_level_below + len(level_candidates) > self.limit:
                 cutoff = self.limit - currently_active_level_below
                 fitness_cutoff = level_candidates[cutoff].fitness
                 for deme in level_demes:
-                    candidates[deme] = (
-                        candidates[deme][0],
-                        list(filter(lambda ind: ind.fitness < fitness_cutoff, candidates[deme][1])),  # type: ignore
-                    )
+                    candidates[deme].individuals = [
+                        ind for ind in candidates[deme].individuals if ind.fitness < fitness_cutoff  # type: ignore
```

Comment:
do we need `# type: ignore`?

Reply:
Unfortunately yes. The precommit fails without it for some reason

Reply:
ok, I thought that the filter caused this issue